lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -1001,55 +1007,34 @@ namespace Js
+                if (!childModuleRecord->WasEvaluationPrepassed())
                 {
-                    if (!childModuleRecord->WasEvaluationPrepassed())
-                    {
-                        childModuleRecord->ModuleEvaluationPrepass();
-                    }
- 
-                    // if child module was evaluated before and threw need to re-throw now
-                    // if child module has been dynamically imported and has exception need to throw
-                    if (childModuleRecord->GetErrorObject() != nullptr)
-                    {
-                        this->ReleaseParserResourcesForHierarchy();
+                    childModuleRecord->ModuleEvaluationPrepass();
+                }
 
-                        JavascriptExceptionOperators::Throw(childModuleRecord->GetErrorObject(), this->scriptContext);
-                    }
-                });
-            }
+                // if child module was evaluated before and threw need to re-throw now
+                // if child module has been dynamically imported and has exception need to throw
+                if (childModuleRecord->GetErrorObject() != nullptr)
+                {
+                    this->ReleaseParserResourcesForHierarchy();
 
-            AUTO_NESTED_HANDLED_EXCEPTION_TYPE((ExceptionType)(ExceptionType_OutOfMemory | ExceptionType_JavascriptException));
-            BEGIN_SAFE_REENTRANT_CALL(scriptContext->GetThreadContext())
-            {
-                Arguments outArgs(CallInfo(CallFlags_Value, 0), nullptr);
-                this->generator = VarTo<JavascriptGenerator>(rootFunction->CallRootFunction(outArgs, scriptContext, true));
-            }
-            END_SAFE_REENTRANT_CALL
-        }
-        catch (const Js::JavascriptException &err)
-        {
-            exception = err.GetAndClear();
-            Var errorObject = exception->GetThrownObject(scriptContext);
-            AssertOrFailFastMsg(errorObject != nullptr, "ModuleEvaluation: null error object thrown from root function");
-            this->errorObject = errorObject;
-            if (this->promise != nullptr)
-            {
-                ResolveOrRejectDynamicImportPromise(false, errorObject, scriptContext, this, false);
-                return false;
-            }
+                    JavascriptExceptionOperators::Throw(childModuleRecord->GetErrorObject(), this->scriptContext);
+                }
+            });
         }
 
-        if (exception != nullptr)
+        AUTO_NESTED_HANDLED_EXCEPTION_TYPE((ExceptionType)(ExceptionType_OutOfMemory | ExceptionType_JavascriptException));
+        BEGIN_SAFE_REENTRANT_CALL(scriptContext->GetThreadContext())
         {
-            JavascriptExceptionOperators::DoThrowCheckClone(exception, scriptContext);
+            Arguments outArgs(CallInfo(CallFlags_Value, 0), nullptr);
+            this->generator = VarTo<JavascriptGenerator>(rootFunction->CallRootFunction(outArgs, scriptContext, true));
```

Comment:
This call executes the "header" of the module which hoists any function exports. It yields before the body of the module.